torchgeo/datasets/oscd.py

```diff
@@ -0,0 +1,313 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""OSCD dataset."""
+
+import glob
+import os
+from typing import Callable, Dict, List, Optional, Sequence, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.figure import Figure
+from numpy import ndarray as Array
+from PIL import Image
+from torch import Tensor
+
+from ..datasets.utils import draw_semantic_segmentation_masks
+from .geo import VisionDataset
+from .utils import download_url, extract_archive, sort_sentinel2_bands
+
+
+class OSCD(VisionDataset):
+    """OSCD dataset.
+
+    The `Onera Satellite Change Detection <https://rcdaudt.github.io/oscd/>`_
+    dataset addresses the issue of detecting changes between
+    satellite images from different dates. Imagery comes from
+    Sentinel-2 which contains varying resolutions per band.
+
+    Dataset format:
+
+    * images are 13-channel tifs
+    * masks are single-channel pngs where no change = 0, change = 255
+
+    Dataset classes:
+
+    0. no change
+    1. change
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1109/IGARSS.2018.8518015
+
+    .. versionadded:: 0.2
+    """
+
+    folder_prefix = "Onera Satellite Change Detection dataset - "
+    urls = {
+        "Onera Satellite Change Detection dataset - Images.zip": (
+            "https://partage.imt.fr/index.php/s/gKRaWgRnLMfwMGo/download"
+        ),
+        "Onera Satellite Change Detection dataset - Train Labels.zip": (
+            "https://partage.mines-telecom.fr/index.php/s/2D6n03k58ygBSpu/download"
+        ),
+        "Onera Satellite Change Detection dataset - Test Labels.zip": (
+            "https://partage.imt.fr/index.php/s/gpStKn4Mpgfnr63/download"
+        ),
+    }
+
+    md5 = "7383412da7ece1dca1c12dc92ac77f09"
```

Comment:
This dataset downloads 3 different zip files but uses the same MD5 for all of them. This is a bug, right? Going to try to fix this in #329 since I need to regenerate the data anyway.

Reply:
What I'm really curious about is the fact that the tests still pass even with the wrong checksum. Even if I change the zip file entirely it still passes the tests.